Plugin Server for Openplanet Plugin

skiff opened the pull request:
- Main plugin server for managing requests/responses from Openplanet plugin
  - Opens port 27990 for TCP connections, will require assigning an IP in ECS
- Additional event bus types for players joining and leaving queues

skiff committed:
infra/docker changes

skiff committed:
initial requests and response base

skiff committed:
add plugin request responses

skiff committed:
events

skiff committed:
working full flow

skiff committed:
add some error handling, always respond to plugin if message is received

skiff committed:
add join and leave queue events

skiff committed:
pre-commit cleanup

skiff committed:
change leave queue to also run a queue update, graceful shutdown plugin connections

Nixotica commented:
Some things to discuss:

- How is a canceled match handled? I tried a solo queue match, wasn't able to join (due to school mode) then canceled it, which the plugin never picked up. This seems like it'll get players stuck in a fake match in the plugin. This probably should be a signal (or it can be used with NEW_COMPLETED_MATCH signal, with `cancel = True`) 
- In the plugin -- Refresh Leaderboards is failing 
```
[   ScriptRuntime] [ TRAC] [18:34:23] [TMOpenMatchmakingBotPlugin]  Sending message: {"Command":"GetLeaderboards","Payload":{},"User":"2e34c3cb-9548-4815-aee3-c68518a1fd88"}
[   ScriptRuntime] [ TRAC] [18:34:23] [TMOpenMatchmakingBotPlugin]  Received message: {"Command": "GetLeaderboardsResponse", "Payload": {"Leaderboards": [{"Id": null, "Self": {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 1000}, "Players": [{"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 1000}]}, {"Id": "\ud83c\udf0e  Global", "Self": {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 214}, "Players": [{"TmAccountId": "6e3bf3f9-7dcb-47d4-bdae-037ab66628f2", "Points": 1009}, {"TmAccountId": "9851b4af-d6f1-4aa9-8fe4-cbcbfb05f2b4", "Points": 1004}, {"TmAccountId": "fec97636-5c56-4c39-a230-4a27143ea3d8", "Points": 993}, {"TmAccountId": "8e0049ea-1145-4440-9ea0-0c0a9c89da05", "Points": 992}, {"TmAccountId": "c7818ba0-5e85-408e-a852-f658e8b90eec", "Points": 992}, {"TmAccountId": "df9448f1-a8d5-4682-9003-1c2777c62b91", "Points": 991}, {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 214}, {"TmAccountId": "ec0269d5-2d19-41eb-a931-01b8a11c2784", "Points": 202}, {"TmAccountId": "febeaf4d-f340-4954-9fe2-88390959ae53", "Points": 17}]}, {"Id": "Unranked", "Self": {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 36}, "Players": [{"TmAccountId": "2232c721-f215-4036-b28b-772eee46632c", "Points": 1019}, {"TmAccountId": "9851b4af-d6f1-4aa9-8fe4-cbcbfb05f2b4", "Points": 976}, {"TmAccountId": "febeaf4d-f340-4954-9fe2-88390959ae53", "Points": 971}, {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 36}]}]}, "StatusCode": 200}
[   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]  Script exception: Json value is not a string!
[   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]    C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Net/Responses.as (line 105, column 13)
[   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]      #0  bool Net::ProcessGetLeaderboards(Json::Value payload) (C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Net/Responses.as line 105)
[   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]      #1  bool Net::ProcessResponse(Json::Value@ msg) (C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Net/Responses.as line 12)
[   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]      #2  void Net::Loop() (C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Net/Net.as line 50)
[   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]      #3  void Main() (C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Main.as line 5)
```

Nixotica committed:
Feat/1v1queue (#87)

* wip 1v1 mode - needs testing

* tested with liquid <3

skiff committed:
add event_name to 1v1 events

skiff committed:
add match canceled command, get leaderboards from active queues instead of database

skiff committed:
Merge branch 'master' into feature/plugin-v2

skiff committed:
fix merges

skiff commented:
> Some things to discuss:
> 
> * How is a canceled match handled? I tried a solo queue match, wasn't able to join (due to school mode) then canceled it, which the plugin never picked up. This seems like it'll get players stuck in a fake match in the plugin. This probably should be a signal (or it can be used with NEW_COMPLETED_MATCH signal, with `cancel = True`)
> * In the plugin -- Refresh Leaderboards is failing
> 
> ```
> [   ScriptRuntime] [ TRAC] [18:34:23] [TMOpenMatchmakingBotPlugin]  Sending message: {"Command":"GetLeaderboards","Payload":{},"User":"2e34c3cb-9548-4815-aee3-c68518a1fd88"}
> [   ScriptRuntime] [ TRAC] [18:34:23] [TMOpenMatchmakingBotPlugin]  Received message: {"Command": "GetLeaderboardsResponse", "Payload": {"Leaderboards": [{"Id": null, "Self": {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 1000}, "Players": [{"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 1000}]}, {"Id": "\ud83c\udf0e  Global", "Self": {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 214}, "Players": [{"TmAccountId": "6e3bf3f9-7dcb-47d4-bdae-037ab66628f2", "Points": 1009}, {"TmAccountId": "9851b4af-d6f1-4aa9-8fe4-cbcbfb05f2b4", "Points": 1004}, {"TmAccountId": "fec97636-5c56-4c39-a230-4a27143ea3d8", "Points": 993}, {"TmAccountId": "8e0049ea-1145-4440-9ea0-0c0a9c89da05", "Points": 992}, {"TmAccountId": "c7818ba0-5e85-408e-a852-f658e8b90eec", "Points": 992}, {"TmAccountId": "df9448f1-a8d5-4682-9003-1c2777c62b91", "Points": 991}, {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 214}, {"TmAccountId": "ec0269d5-2d19-41eb-a931-01b8a11c2784", "Points": 202}, {"TmAccountId": "febeaf4d-f340-4954-9fe2-88390959ae53", "Points": 17}]}, {"Id": "Unranked", "Self": {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 36}, "Players": [{"TmAccountId": "2232c721-f215-4036-b28b-772eee46632c", "Points": 1019}, {"TmAccountId": "9851b4af-d6f1-4aa9-8fe4-cbcbfb05f2b4", "Points": 976}, {"TmAccountId": "febeaf4d-f340-4954-9fe2-88390959ae53", "Points": 971}, {"TmAccountId": "2e34c3cb-9548-4815-aee3-c68518a1fd88", "Points": 36}]}]}, "StatusCode": 200}
> [   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]  Script exception: Json value is not a string!
> [   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]    C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Net/Responses.as (line 105, column 13)
> [   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]      #0  bool Net::ProcessGetLeaderboards(Json::Value payload) (C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Net/Responses.as line 105)
> [   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]      #1  bool Net::ProcessResponse(Json::Value@ msg) (C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Net/Responses.as line 12)
> [   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]      #2  void Net::Loop() (C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Net/Net.as line 50)
> [   ScriptRuntime] [ERROR] [18:34:23] [TMOpenMatchmakingBotPlugin]      #3  void Main() (C:/Users/nixot/OpenplanetNext/Plugins/TMOpenMatchmakingBotPlugin/src/Main.as line 5)
> ```

Added some logic for handling canceled matches. Will reset the plugin back to initial state if a players match is canceled,. Updated the refresh leaderboards logic to only pull back the leaderboards based on the active queues instead of all leaderboards in the database table

Nixotica commented:
Looks good, we can merge this
